control/plugin/session.go

```diff
@@ -57,9 +57,12 @@ type Session interface {
 	isDaemon() bool
 
 	SetKey(SetKeyArgs, *[]byte) error
+	setKey([]byte)
```

Comment:
Just wanted to make a note. I think this `setKey` and `SetKey` differentiation is a little dense.  Because a gRPC plugin will not need encoding or payload encryption, I think we should rethink this interface; maybe a way to compose interfaces depending on RPC type.